pkgs/_analyzer_macros/bin/server.dart

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:_analyzer_macros/macro_implementation.dart';
+import 'package:analysis_server/starter.dart';
+import 'package:analyzer/src/summary2/macro_injected_impl.dart' as injected;
+import 'package:macro_service/macro_service.dart';
+
+/// Analysis server with `dart_model` implementation injected.
+///
+/// Run with your IDE by compiling and placing in the SDK your IDE is using,
+/// for example:
+///
+/// dart compile kernel bin/server.dart
+/// cp bin/server.dill ~/opt/dart-sdk-be/bin/snapshots/analysis_server.dart.snapshot
+///
+/// Then restart, VSCode: Ctrl+Shift+P, Restart Analysis Server.
+///
+/// Only works for one path, see `packageConfig` path hardcoded below.
+void main(List<String> args) async {
+  final home = Platform.environment['HOME']!;
+  injected.macroImplementation = await AnalyzerMacroImplementation.start(
+      protocol: Protocol(
+          encoding: ProtocolEncoding.binary, version: ProtocolVersion.macros1),
+      // TODO(davidmorgan): this needs to come from the analyzer, not be
+      // hardcoded.
+      packageConfig:
+          Uri.file('$home/git/macros/.dart_tool/package_config.json'));
```

Comment:
I believe this can be `Isolate.packageConfig` for now? At least it won't be specific to your file system layout then 😄

Reply:
Actually nvm I think there won't be a package config when this actually runs.

However, maybe a `-D` variable, then the path gets compiled into the script as a constant.

Reply:
Right.

`-D` sounds good; done and updated the example command.